src/algokit/cli/deploy.py

```diff
@@ -0,0 +1,108 @@
+import logging
+import os
+from pathlib import Path
+
+import click
+
+from algokit.core.constants import DEFAULT_NETWORKS, DEPLOYER_KEY, DISPENSER_KEY, LOCALNET, MAINNET
+from algokit.core.deploy import (
+    execute_deploy_command,
+    get_genesis_network_name,
+    load_deploy_command,
+    load_deploy_config,
+)
+
+logger = logging.getLogger(__name__)
+
+
+def extract_mnemonics(skip_mnemonics_prompts: bool, network: str) -> tuple[str | None, str | None]:  # noqa: FBT001
+    """
+    Extract environment variables, prompt user if needed.
+
+    :param skip_mnemonics_prompts: A boolean indicating if user prompt should be skipped.
+    :param network: The name of the network ('localnet', 'testnet', 'mainnet', etc.)
+    :return: A tuple containing deployer_mnemonic and dispenser_mnemonic.
+    """
+    deployer_mnemonic = os.environ.get(DEPLOYER_KEY, "")
+    dispenser_mnemonic = os.environ.get(DISPENSER_KEY, None)
+
+    is_not_localnet = network != LOCALNET
+    is_deployer_mnemonic_empty = not skip_mnemonics_prompts and not deployer_mnemonic
+    is_dispenser_mnemonic_empty = not skip_mnemonics_prompts and not dispenser_mnemonic
+
+    if is_deployer_mnemonic_empty and is_not_localnet:
+        deployer_mnemonic = click.prompt("deployer-mnemonic", hide_input=True)
+
+    if (
+        is_dispenser_mnemonic_empty
+        and is_not_localnet
+        and click.confirm("Do you want to use a dispenser account?", default=False)
+    ):
+        dispenser_mnemonic = click.prompt("dispenser-mnemonic", hide_input=True)
+
+    return deployer_mnemonic, dispenser_mnemonic
+
+
+@click.command("deploy")
+@click.argument(
+    "network", type=str, default=LOCALNET, required=True, callback=(lambda _ctx, _param, value: value.lower())
```

Comment:
I had to look up what `callback` does to be sure - wouldn't it be simpler to have the `.lower()` call inside the command function?
also `type=str` is the default
also, what does it mean for an argument to have a default but also be required?

Reply:
You are right, required isnt needed since we assume localnet by default - this should be fixed now and im using lower explicitly inside the function itself instead of the callback